helm.el

```diff
@@ -3114,12 +3119,35 @@ and `helm-pattern'."
              (if (string-match "[[:upper:]]" pattern) nil t)))
     (t helm-case-fold-search)))
 
-(defun helm-match-from-candidates (cands matchfns match-part-fn limit source)
-  (let (matches)
+
+(defun helm--mapconcat-initials-pattern (pattern seperators)
```

Comment:
What you're suggesting I've already covered in the original issue #1237, please read it.
Also, Inserting greedy matching operators can and will drop the best match from the list.

> The more candidates flx sees the better your results will be.

Right, but that I've had to set the limit at 2500 at one point to get the match I wanted,
it was acceptable but not very quick, It caused long GCs  (yes, I know about the
GC tuning you recommend) and most of that work is completely unnecessary. 
What you're suggesting